### DISCUSSION TITLE
How to connect to database to run some select queries

### DISCUSSION THREAD

**Discussion by jlmanohar** [nextcloud/all-in-one#1728] [Questions] [answered]
Hi, i have an issue where the recognize app added some files into queue for object detection, but has not progressed, i want to access the postgress db to run some queries to check if there is lock on any tables that it uses... and to see what are in queue to maybe manually remove them.

How do i connect to the DB to execute those select queries?
![image](https://user-images.githubusercontent.com/66161434/212057277-99ffc005-9fa1-426e-81cc-bc0fa7a23d75.png)

**Comment by szaimen** [comment, marked as answer]
Hi, you can go into the database container with `sudo docker exec -it nextcloud-aio-database bash` and run some queries then via CLI. For the exact syntax, best is to refer to postgresql docs or recognize docs. Datbasae user is oc_nextcloud, database name nextcloud_database, passwordd `$POSTGRES_PASSWORD`.

**Reply by helmo** [reply]
I was also looking for this, and came up with:

```
echo $POSTGRES_PASSWORD

psql -p 5432 -h 127.0.0.1 --username=$POSTGRES_USER --dbname=$POSTGRES_DB --password
```

The output of the first echo should then be pasted into the 'Password:' prompt.

**Reply by th0rgall** [reply]
Connecting is also possible in one line, without copy-pasting: 
```bash
docker exec -it nextcloud-aio-database bash -c 'PGPASSWORD=$POSTGRES_PASSWORD psql -p 5432 -h 127.0.0.1 --username=$POSTGRES_USER --dbname=$POSTGRES_DB'
```
